sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -317,6 +317,10 @@ public static void rename(
       return;
     }
 
+    // XXX: Instead of filtering here we should pass options on to filesystem as it will be able to
```

Comment:
This is a TODO ?

Reply:
I'm leaving XXX for things I don't want to submit

I am pushing this for a preliminary review that it doesn't seem broken and then so we can load test with it.  It definitely has warts that need to be addressed before submiting but we can do that if it turns out to be beneficial.

Reply:
Fixed all XXXs